### ISSUE TITLE
Can't Compile/Configure on Ubuntu 20.04 LTS

### ISSUE THREAD

**Issue by act17** [FWGS/hlsdk-portable#380]
When trying to configure through `cmake -B build -S .`, I get the following output:
```
-- Barnacle tongue length fix is disabled
-- Client local weapons prediction is enabled
-- Crowbar idle animation is disabled
-- Crowbar attack delay fix is disabled
-- Rapid crowbar fix is disabled
-- Gauss overcharge fix is disabled
-- Fix of tripmine beam duplication on level transition is disabled
-- Handgrenade deploy animation fix after finishing a throw is disabled
-- Animation times fix for some weapons is disabled
-- OEM Build is disabled
-- Demo Build is disabled
CMake Error at cmake/LibraryNaming.cmake:111 (message):
  Place your operating system name here! If this is a mistake, try to fix
  conditions above and report a bug
Call Stack (most recent call first):
  CMakeLists.txt:110 (include)


CMake Error at cmake/LibraryNaming.cmake:172 (message):
  Place your architecture name here! If this is a mistake, try to fix
  conditions above and report a bug
Call Stack (most recent call first):
  CMakeLists.txt:110 (include)


-- Library postfix: 
-- Building for 32 Bit
-- Building client enabled
-- Building server enabled
-- Configuring incomplete, errors occurred!
See also "/home/aiden/git/hlsdk-portable/build/CMakeFiles/CMakeOutput.log".
See also "/home/aiden/git/hlsdk-portable/build/CMakeFiles/CMakeError.log".
```

When trying to configure through `./waf configure -8`, I get the following output:
```
Setting top to                                               : /home/aiden/git/hlsdk-portable 
Setting out to                                               : /home/aiden/git/hlsdk-portable/build 
Build type                                                   : release 
LTO build                                                    : no 
PolyOpt build                                                : no 
Checking for 'gcc' (C compiler)                              : /usr/bin/gcc 
Target OS                                                    : linux 
Target CPU                                                   : x86_64 
Target binfmt                                                : elf 
Checking for 'g++' (C++ compiler)                            : /usr/bin/g++ 
Target OS                                                    : linux 
Target CPU                                                   : x86_64 
Target binfmt                                                : elf 
Checking for program 'strip'                                 : /usr/bin/strip 
Checking for program 'objcopy'                               : /usr/bin/objcopy 
Checking if 'gcc' can target 32-bit                          : no 
Checking for required C flags                                : yes 
Checking for required C++ flags                              : yes 
Checking supported flags for gcc in parallel                 : started 
... -fnonconst-initializers                                  : no 
... -Werror=packed                                           : yes 
... -Werror=bool-operation                                   : yes 
... -Werror=enum-conversion                                  : no 
... -Werror=implicit-fallthrough=2                           : yes 
... -Werror=bool-compare                                     : yes 
... -Werror=sizeof-pointer-memaccess                         : yes 
... -Wunintialized                                           : no 
... -Werror=return-type                                      : yes 
... -Winit-self                                              : yes 
... -Werror=vla                                              : yes 
... -Werror=cast-align=strict                                : yes 
... -Werror=packed-not-aligned                               : yes 
... -Werror=jump-misses-init                                 : yes 
... -Werror=sizeof-array-div                                 : no 
... -Werror=tautological-compare                             : yes 
... -Werror=implicit-int                                     : yes 
... -Werror=duplicated-cond                                  : yes 
... -Wmisleading-indentation                                 : yes 
... -Werror=parentheses                                      : yes 
... -Werror=incompatible-pointer-types                       : yes 
... -Werror=use-after-free=3                                 : no 
... -Werror=sequence-point                                   : yes 
... -Werror=declaration-after-statement                      : yes 
... -Werror=write-strings                                    : yes 
... -Werror=int-conversion                                   : yes 
... -Werror=implicit-function-declaration                    : yes 
... -Werror=sizeof-pointer-div                               : yes 
... -Werror=string-compare                                   : no 
... -fdiagnostics-color=always                               : yes 
-> processing test results                                   : 6 test failed 
Checking supported flags for g++ in parallel                 : started 
... -Werror=string-compare                                   : no 
... -Winit-self                                              : yes 
... -Werror=bool-operation                                   : yes 
... -Werror=sizeof-pointer-div                               : yes 
... -Werror=packed-not-aligned                               : yes 
... -Werror=vla                                              : yes 
... -Werror=tautological-compare                             : yes 
... -Werror=implicit-fallthrough=2                           : yes 
... -Werror=sizeof-pointer-memaccess                         : yes 
... -Werror=duplicated-cond                                  : yes 
... -Wmisleading-indentation                                 : yes 
... -Werror=sizeof-array-div                                 : no 
... -Werror=use-after-free=3                                 : no 
... -Werror=parentheses                                      : yes 
... -Wunintialized                                           : no 
... -Werror=sequence-point                                   : yes 
... -Werror=write-strings                                    : yes 
... -Werror=return-type                                      : yes 
... -Werror=cast-align=strict                                : yes 
... -Werror=bool-compare                                     : yes 
... -Werror=packed                                           : yes 
... -fdiagnostics-color=always                               : yes 
-> processing test results                                   : 4 test failed 
Checking for library m                                       : yes 
Checking for header tgmath.h                                 : yes 
Checking if tgmath.h is usable                               : yes 
Checking if cmath is usable                                  : yes 
-> processing mod options                                    : ... 
Traceback (most recent call last):
  File "/home/aiden/git/hlsdk-portable/.waf-2.0.25-9dbe1df4d0e92facbc1de0486ec78fed/waflib/Scripting.py", line 163, in waf_entry_point
    run_commands()
  File "/home/aiden/git/hlsdk-portable/.waf-2.0.25-9dbe1df4d0e92facbc1de0486ec78fed/waflib/Scripting.py", line 259, in run_commands
    ctx = run_command(cmd_name)
  File "/home/aiden/git/hlsdk-portable/.waf-2.0.25-9dbe1df4d0e92facbc1de0486ec78fed/waflib/Scripting.py", line 243, in run_command
    ctx.execute()
  File "/home/aiden/git/hlsdk-portable/.waf-2.0.25-9dbe1df4d0e92facbc1de0486ec78fed/waflib/Configure.py", line 163, in execute
    super(ConfigurationContext, self).execute()
  File "/home/aiden/git/hlsdk-portable/.waf-2.0.25-9dbe1df4d0e92facbc1de0486ec78fed/waflib/Context.py", line 218, in execute
    self.recurse([os.path.dirname(g_module.root_path)])
  File "/home/aiden/git/hlsdk-portable/.waf-2.0.25-9dbe1df4d0e92facbc1de0486ec78fed/waflib/Context.py", line 300, in recurse
    user_function(self)
  File "/home/aiden/git/hlsdk-portable/wscript", line 219, in configure
    conf.start_msg("* " + p[3])
TypeError: '_sre.SRE_Match' object has no attribute '__getitem__'
```

**Comment by act17** [comment]
I have no idea whats going on with the formatting, I'm sorry.

**Comment by a1batross** [comment]
I fixed it for you. For code blocks you must use triple `

**Comment by act17** [comment]
Thanks big chief, I'll remember that for the future.

**Comment by a1batross** [comment]
I just tried it in raw Ubuntu 20.04 container.

Yeah, the message is a bit cryptic if multilib isn't installed. The worse thing is that after installing it, CMake won't fix itself, you need to clean build directory.

Also, mod_options parsing is broken in wscript on Python 2... Will fix it later. :(

**Comment by act17** [comment]
Alright, after installing the multilib packages, I can configure. However, when building, I get the following error:
```
[ 25%] Building CXX object cl_dll/CMakeFiles/client.dir/input_goldsource.cpp.o
In file included from /usr/include/SDL2/SDL_stdinc.h:31,
                 from /usr/include/SDL2/SDL_mouse.h:31,
                 from /home/aiden/git/hlsdk-portable/cl_dll/input_goldsource.cpp:33:
/usr/include/SDL2/SDL_config.h:4:10: fatal error: SDL2/_real_SDL_config.h: No such file or directory
    4 | #include <SDL2/_real_SDL_config.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [cl_dll/CMakeFiles/client.dir/build.make:596: cl_dll/CMakeFiles/client.dir/input_goldsource.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:115: cl_dll/CMakeFiles/client.dir/all] Error 2
```

``libsdl2-dev`` is already installed.

**Comment by a1batross** [comment]
libsdl2-dev:i386?

You need exactly this.

**Comment by act17** [comment]
```sudo apt install libsdl2-dev:i386
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help resolve the situation:

The following packages have unmet dependencies:
 libsdl2-dev:i386 : Depends: libasound2-dev:i386 but it is not going to be installed
                    Depends: libibus-1.0-dev:i386 but it is not going to be installed
                    Depends: libpulse-dev:i386 but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```
send help

**Comment by a1batross** [comment]
Oh, it's probably 20.04. Some old Ubuntu versions had bug where installing 32-bit dev version of SDL2 required removing 64-bit packages.

Either upgrade your distro, or target 64-bit during development of your mod.

**Comment by act17** [comment]
How lovely. How would I target 64-bit when compiling this?

**Comment by a1batross** [comment]
In CMake it's -D64BIT=1, in Waf it's -8, that you already used.

**Comment by FreeSlave** [comment]
> Either upgrade your distro, or target 64-bit during development of your mod.

Or just follow the instructions (or github actions) and use Steam Runtime or another 32-bit chroot.

**Comment by MusaHabila84** [comment]
WERROR implicit declaration of function

**Comment by a1batross** [comment]
What? Can you describe your problem?

**Comment by a1batross** [comment]
Anyway, this issue is probably resolved now.
